### PR TITLE
fix: added failsafes to the animate functions based on evaluations

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Added failsafe evaluations to the animate functions to prevent execution if refs don't exist/render.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-809 #done

### CODE REVIEW

File: packages/shared/src/components/RankProgress.tsx [hunk 1]
```diff
@@ -162,74 +162,80 @@ export function RankProgress({
     const firstAnimationDuration = 400;
     const maxScale = 1.666;
 
-    const progressAnimation = showRadialProgress
-      ? progressRef.current.animate(
```
Comment (nensidosari) [inline packages/shared/src/components/RankProgress.tsx:166]: Probably the optional chaining would work well here too

File: packages/shared/src/components/RankProgress.tsx [hunk 2]
```diff
@@ -162,74 +162,80 @@ export function RankProgress({
...
-                  transform: 'scale(1.5)',
-                  opacity: 0,
+                  transform: 'scale(1)',
+                  opacity: 1,
+                  '--stop-color1': rankToGradientStopBottom(rank),
+                  '--stop-color2': rankToGradientStopTop(rank),
                 },
               ],
-              { duration: 600, fill: 'forwards' },
+              { duration: 100, fill: 'forwards' },
```
Comment (nensidosari) [inline packages/shared/src/components/RankProgress.tsx:236]: Is this change intentional?

Comment (rebelchris) [reply packages/shared/src/components/RankProgress.tsx:236]: Oh no! not even sure how that happened 🤯

Comment (rebelchris) [reply packages/shared/src/components/RankProgress.tsx:236]: Ah It's actually not a change, it's just GitHub showing it on the wrong line.
This is the lastBadge which always had 100, it's showing it compared to AttentionRef though.

File: packages/shared/src/components/RankProgress.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/RankProgress.tsx:166]: Although would probably work, I think this is safer down the line to exclude any weird things.
Let me know what you think?

Comment (nensidosari) [reply packages/shared/src/components/RankProgress.tsx:166]: Well that is quite useful in this cases at least when you're just checking down the chain if any of the parts is null/undefined and can make code clearer in this case. I try to use when I can.
But don't have any 100% strong opinion of using either way if we don't have any standard set.
In this case for example with it would allow you to add an extra check for progressRef as well `progressRef?.current?.animate()`

File: packages/shared/src/components/RankProgress.tsx (hunk 2 above)
Comment (nensidosari) [reply packages/shared/src/components/RankProgress.tsx:236]: Ah yeah, now I see. Sometimes GitHub messes with the changes like this 🤦‍♀️ Nevermind then. :D

**Review by nensidosari — APPROVED** [review]
LGTM :) minor suggestion only

File: packages/shared/src/components/RankProgress.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/RankProgress.tsx:166]: Alright, moving to optional chaining :) 
We don't need the `?.current` thought as it will exist as we use refs.
